src/widget/city_building_ghost.c

```diff
@@ -453,8 +454,20 @@ static void draw_default(const map_tile *tile, int x_view, int y_view, building_
                 forbidden_terrain &= ~TERRAIN_WALL;
             }
         }
-        if (fully_blocked || forbidden_terrain || (map_has_figure_at(tile_offset) && type != BUILDING_PLAZA)) {
+
+        if (fully_blocked || forbidden_terrain)
+        {
+            blocked_tiles[i] = 1;
+        } else if (map_has_figure_at(tile_offset) && type != BUILDING_PLAZA) {
             blocked_tiles[i] = 1;
+            int figure_id = map_figure_at(tile_offset);
+            figure *f = figure_get(figure_id);
+            if (f->type == FIGURE_SHEEP || f->type == FIGURE_ZEBRA) {
```

Comment:
I think it'd be better for the logical division to make animal fleeing a function in `figuretype/animal.c` and call it here.

Reply:
Agreed